ansible_bender/callback_plugins/snapshoter.py

```diff
@@ -3,6 +3,9 @@
 import logging
 import os
 import traceback
+import re
+
+from checksumdir import dirhash
```

Comment:
I'm not too fond of including a dependency for sake of tenths of lines of code: https://github.com/cakepietoast/checksumdir/blob/master/checksumdir/__init__.py#L29

the readme also says the library ignores file names:

    A simple module for creating a single hash for a directory of files, with file contents; ignoring any metadata such as file name.

I think a better (and more known) approach would be to check modification time of all files deterministically and hash that, since that's how caching mechanism in docker works

Reply:
I've removed the checksumdir dependency. This is the function I use to get a directory fingerprint
```
    @staticmethod
    def get_dir_fingerprint(directory):
        sha512 = hashlib.sha512()
        for filename in os.listdir(directory):
            full_path = os.path.join(directory, filename)
            the_file = pathlib.Path(full_path)

            if(not(the_file.exists())):
                continue

            file_stats = str(the_file.stat())
            sha512.update(file_stats.encode("utf-8"))

        return sha512.hexdigest()
```

Reply:
Oh dammit, this doesn't handle multi level directories. I'll have to tinker some more.

Reply:
Hopefully final version of get_dir_fingerprint()
```
    @staticmethod
    def get_dir_fingerprint(directory):
        sha512 = hashlib.sha512()
        for root, dirs, files in os.walk(directory):
            for filename in files:
                full_path = os.path.join(directory, root, filename)
                the_file = pathlib.Path(full_path)

                if(not(the_file.exists())):
                    continue

                file_stats = str(the_file.stat())
                sha512.update(file_stats.encode("utf-8"))
```